app/src/main/java/co/runloop/influencer/fragment/ContactsFragment.java

```diff
@@ -54,21 +58,37 @@ public View onCreateView(@NonNull LayoutInflater inflater,
         recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
         recyclerView.setAdapter(adapter);
 
+        progressBar = root.findViewById(R.id.frag_contacts_list_progressbar);
+
         contactsViewModel = ViewModelProviders
                 .of(this)
                 .get(ContactsViewModel.class);
-        contactsViewModel.getContacts().observe(this, (@Nullable List<Contact> contacts) -> {
-            adapter.setContacts(contacts);
-            adapter.notifyDataSetChanged();
-        });
-        if (contactsIsAvailable()) {
-            contactsViewModel.onPermissionGranted();
-            if(contactsViewModel.getContacts().getValue() == null) {
-                contactsViewModel.loadAll();
+        contactsViewModel.getContacts().observe(this, (@Nullable Resource<List<Contact>> contacts) -> {
```

Comment:
We might wanna rename `contacts` to `contactsResource` (or something similar) so it is clearer that it's not actually the list of contacts (since calls like `contacts.getError()` are somewhat not intuitive)

Reply:
i agree